pkg/furnace/furnace.go

```diff
@@ -32,36 +29,25 @@ func RegisterFurnaceServer(s *grpc.Server, fc *Furnace) {
 	api.RegisterFurnaceServer(s, fc)
 }
 
-// Enqueue adds a package from the build queue.
-func (f *Furnace) enqueue(packages []Package) {
-	f.mx.Lock()
-	defer f.mx.Unlock()
-	f.buildQueue = append(f.buildQueue, packages...)
-}
-
-// Dequeue removes a package from the build queue.
-// func (f *Furnace) dequeue() Package {
-// 	f.mx.Lock()
-// 	defer f.mx.Unlock()
-// 	pkg := f.buildQueue[0]
-// 	f.buildQueue = f.buildQueue[1:]
-// 	return pkg
-// }
-
 func (f *Furnace) Build(ctx context.Context, req *api.BuildRequest) (*api.BuildResponse, error) {
 	packages := make([]Package, len(req.GetPackages()))
 	for i, v := range req.GetPackages() {
-		packages[i] = Package{Name: v.GetName(), Version: v.GetVersion()}
+		packages[i] = Package{Name: v.GetName()}
 	}
 	// TODO(prmsrswt): Check if any on these packages is already queued.
-	f.enqueue(packages)
+	f.buildQueue.enqueue(packages)
```

Comment:
Is the final way in which were adding elements for he queue ? Had plans for multiprocessing queues right? Or we improving later ?

Reply:
For multi-processing queues, we just need to consume it from multiple places, and that's still possible with it.